kong/resty/dns/client.lua

```diff
@@ -838,39 +811,45 @@ end
 local function syncQuery(qname, r_opts, try_list)
   local key = qname..":"..r_opts.qtype
   local item = queue[key]
+  local is_yieldable = in_yieldable_phase()
 
-  -- if nothing is in progress, we start a new async query
-  if not item then
-    local err
-    item, err = asyncQuery(qname, r_opts, try_list)
+  -- If not yieldable, we start a new async query and return nil
+  if not is_yieldable then
     if not item then
-      return item, err, try_list
+      asyncQuery(qname, r_opts, try_list)
     end
-  else
-    add_status_to_try_list(try_list, "in progress (sync)")
+    -- item.result is definitely nil in this situation
+    return nil, "API disabled in the unyieldable phase", try_list
```

Comment:
TIPS: [The original logic](https://github.com/Kong/lua-resty-dns-client/pull/130/files#diff-c220f0318ab8cbd55a2893ac929b5b0337a36c8046eb52d2e11e9880f985583bR842) returns `items + nil`, it breaks the convention of syncQuery() API, in its comment, we know that it only returns `item.record, nil` or `nil, err`.

But this wrong return value `item` can be handled by [its caller `resolver()` ](https://github.com/Kong/kong/blob/master/kong/resty/dns/client.lua#L1251), however, `syncQuery()` is not exported from this library, so the user of this library will not notice this wrong logic.

Reply:
"item" is essentially a future object here. IMO, the library's interface should be refactored for all uses "future" and users may have the right expectation about how it could be used (after all future does not look like working with init phase).

Reply:
OK I will add this improvement to our future ticket to refactor this lib